run-tests.sh

```diff
@@ -71,6 +72,26 @@ for arch in $ARCHS; do
         for test in $TESTS_SSP; do
             $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test.exe -fstack-protector-strong
         done
+        for test in $TESTS_UWP; do
+            set +e
+            # compilation should fail for UWP and WinRT
+            $arch-w64-$target_os-clang $test.c -o $TEST_DIR/$test.exe -Wimplicit-function-declaration -Werror
+            UWP_ERROR=$?
+            set -e
+            case $target_os in
+            mingw32uwp|mingw32winrt)
+                if [ $UWP_ERROR -eq 0 ]; then
+                    echo "UWP compilation should have failed for test $test!"
+                    exit 1
+                fi
+                ;;
+            *)
+                if [ $UWP_ERROR -eq 0 ]; then
+                    TESTS_EXTRA="$TESTS_EXTRA $test"
+                fi
```

Comment:
I'd want an `else echo "$test failed to compile for non-UWP target"; exit 1;` here. Otherwise this test could look like it's succeeding (always failing for UWP, and never running the test for non-UWP) if the test source file just is plain broken.

Reply:
Done